packages/tizen_rpc_port/example/server/lib/main.dart

```diff
@@ -0,0 +1,92 @@
+// Copyright 2022 Samsung Electronics Co., Ltd. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// ignore_for_file: public_member_api_docs
+
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+
+import 'message_server.dart';
+
+void main() {
+  runApp(const MyApp());
+}
+
+class Service extends ServiceBase {
+  Service(super.sender, super.instance);
+
+  String _name = '';
+  NotifyCallback? _callback;
+
+  @override
+  Future<void> onCreate() async {
+    print('onCreate. instance: $instance');
+  }
+
+  @override
+  Future<void> onTerminate() async {
+    print('onTerminate. instance: $instance');
+  }
+
+  @override
+  Future<int> onRegister(String name, NotifyCallback callback) async {
+    print('register. instance: $instance, name: $name');
+    _name = name;
+    _callback = callback;
+    return 0;
+  }
+
+  @override
+  Future<int> onSend(String message) async {
+    print('send. instance: $instance, msg: $message');
+    _callback?.invoke(_name, message);
+    return 0;
+  }
+
+  @override
+  Future<void> onUnregister() async {
+    print('unregister. instance: $instance');
+    _callback = null;
+  }
+}
+
+class MyMessageServer extends Message {
+  @override
+  ServiceBase createInstance(String sender, String instance) =>
+      Service(sender, instance);
+}
```

Comment:
You don't have to define a new class. How about changing the definition of `Message` to

```dart
typedef InstanceBuilder = ServiceBase Function(String sender, String instance);

class Message extends StubBase {
  Message({
    required InstanceBuilder instanceBuilder,
  })  : _instanceBuilder = instanceBuilder,
    ...
  }

  final InstanceBuilder _instanceBuilder;
```

And in `main.dart`, the stub can be created like:
```dart
class _MyAppState extends State<MyApp> {
  final Message _messageStub = Message(
    instanceBuilder: (String sender, String instance) {
      return Service(sender, instance);
    },
  );
```

Please refer to the "item builder" pattern used by Flutter: https://docs.flutter.dev/cookbook/lists/long-lists

The builder argument can also be moved to the `listen` method but doing so will require additional refactoring.

Reply:
I applied this suggestion.
Thank you!

Reply:
Should the typedef name have the `Message` prefix? Isn't the signature always the same?